### PR TITLE
[CNVM] Sanity test

### PR DESCRIPTION
**Pull request by amirbenun**
### Summary of your changes
First sanity test to vulnerability management integration:
1. Create an elastic stack on the ESS environment
2. Create agent policy and install CNVM integration
3. Deploy the CloudFormation stack
4. Teardown

### Related Issues
- https://github.com/elastic/cloudbeat/issues/700

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

**Comment by oren-zohar** [comment]
@amirbenun how much time does this flow add to the original test?

**Comment by amirbenun** [comment]
>  how much time does this flow add to the original test?


It adds about a minute so far

image

The second phase of waiting for vulnerabilities might take longer

File: .github/workflows/test-environment.yml [hunk 1]
```diff
@@ -163,6 +164,17 @@ jobs:
           echo "$summary" >> $GITHUB_STEP_SUMMARY
           echo "$summary" # Print the summary to the workflow log
 
+      - name: Install CNVM integration
+        id: cnvm
+        run: |
+          cd fleet_api
+          poetry run python src/install_cnvm_integration.py
+
```
Comment (gurevichdmitry) [inline .github/workflows/test-environment.yml:172]: @amirbenun, please update `fleet_api` README with this command

File: deploy/test-environments/fleet_api/src/install_cnvm_integration.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+"""
+This script installs CNVM AWS integration
+
+The following steps are performed:
+1. Create an agent policy.
+2. Create a CNVM AWS integration.
+3. Create a CNVM bash script to be deployed on a host.
```
Comment (gurevichdmitry) [inline deploy/test-environments/fleet_api/src/install_cnvm_integration.py:8]: This script generates the config.json file, which is utilized in the deployment of the stack (`deploy/cloudformation`)

File: deploy/test-environments/fleet_api/src/install_cnvm_integration.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        )
+    else:
+        cloudformation_params.ELASTIC_ARTIFACTS_SERVER = (
+            cnfg.artifactory_url["staging"]
+            + get_build_info(
+                version=cloudformation_params.ELASTIC_AGENT_VERSION,
+                is_snapshot=False,
+            )
+            + CNVM_ARTIFACT_SUFFIX
+        )
```
Comment (gurevichdmitry) [inline deploy/test-environments/fleet_api/src/install_cnvm_integration.py:100]: @amirbenun,  I’m not sure `ELASTIC_ARTIFACTS_SERVER` is correct
Screenshot 2023-06-27 at 19 36 28

**Review by gurevichdmitry — COMMENTED** [review]
@amirbenun great job! I've reviewed your work and have provided some comments

File: deploy/test-environments/fleet_api/src/api/common_api.py [hunk 4]
```diff
@@ -125,7 +125,7 @@ def get_build_info(version: str, is_snapshot: bool) -> str:
     """
     # pylint: disable=duplicate-code
     if is_snapshot:
-        url = "https://snapshots.elastic.co/latest/master.json"
+        url = f"https://snapshots.elastic.co/latest/{version}.json"
```
[the remark this answers is not in the document]
Comment (amirbenun) [reply deploy/test-environments/fleet_api/src/api/common_api.py:131]: For example right now, after the `8.10` branching out, the master link points to version is `8.10.0-SNAPSHOT` but `8.9.0` was not released yet.
This change is required to be able to test `8.9.0-SNAPSHOT`.

**Review by gurevichdmitry — APPROVED** [review]
@amirbenun great work! approved

**Pull request merged**